lfsapi/endpoint_finder.go

```diff
@@ -26,14 +26,28 @@ const (
 	defaultRemote              = "origin"
 )
 
+type Access struct {
+	mode AccessMode
+	url  string
+}
+
+// Returns a copy of an AccessMode with the mode upgraded to newMode
+func (a *Access) Upgrade(newMode AccessMode) Access {
+	return Access{url: a.url, mode: newMode}
+}
+
+func (a *Access) GetMode() AccessMode {
```

Comment:
Nit: "accessors" in Go do not ordinarily contain the prefix `Get()`. I think that you did the right thing in making `mode` non-exported, since that enforces immutability. But, let's call this `Mode()` instead, to make it appear closer to accessing a normal, publicly exported field.